retrievalmarket/impl/dtutils/dtutils.go

```diff
@@ -50,26 +65,24 @@ func providerEvent(event datatransfer.Event, channelState datatransfer.ChannelSt
 // event or moving to error if a data transfer error occurs
 func ProviderDataTransferSubscriber(deals EventReceiver) datatransfer.Subscriber {
 	return func(event datatransfer.Event, channelState datatransfer.ChannelState) {
-		dealProposal, ok := dealProposalFromVoucher(channelState.Voucher())
-		// if this event is for a transfer not related to storage, ignore
-		if !ok {
+		voucher := channelState.Voucher()
+		if voucher.Voucher == nil {
+			log.Errorf("received empty voucher")
```

Comment:
if we get this in the logs, what is the likelihood that it's going to be in close-enough proximity to other messages that tell us what transfer this is for? do we need to start putting context-specific information in these messages to help with debugging?

Reply:
a good question. ultimately, our pressing priority is going to be to improve error messages returned over the wire, since the data we collect is primarily from the client side. We often already have plenty of information on this side, but minimal ability to access it (cause it's per miner logs).

Let's address this as we move forward.